lib/active_record/connection_adapters/clickhouse_adapter.rb

```diff
@@ -223,6 +228,8 @@ def quote(value)
         case value
         when Array
           '[' + value.map { |v| quote(v) }.join(', ') + ']'
+        when Hash
+          '{' + value.map { |k, v| "#{quote(k)}: #{quote(v)}" }.join(', ') + '}'
```

Comment:
Maybe `value.to_json` will be better?

Reply:
That would definitely be cleaner, however, looking at the [rails source](https://github.com/rails/rails/blob/290b8069b98b87717525382ae2bad5cc2fbf3b66/activerecord/lib/active_record/connection_adapters/abstract/quoting.rb#L73) it looks like manual quoting is more about security than anything.

Tracing the postgres adapter's [implementation](https://github.com/ged/ruby-pg/blob/37197594e64f85cc87322ae8ce16096258bd2c2e/lib/pg/text_encoder/json.rb#L10), it looks like they use `JSON.generate`.

```suggestion
          JSON.generate(value, quirks_mode: true)
```

Reply:
I had to revert this; clickhouse does not always like JSON generated values a67ca2bbec38953f4bf7f5719a58175466c75130